C4/Budgets.pm

```diff
@@ -363,7 +369,12 @@ sub GetBudgetOrdered {
             datecancellationprinted IS NULL
     |);
 	$sth->execute($budget_id);
-	my $sum =  $sth->fetchrow_array;
+    my $sum =  0 + $sth->fetchrow_array;
+
+    my $adjustments = Koha::InvoiceAdjustments->search({budget_id => $budget_id, encumber_open => 1, closedate => undef},{ join => 'invoiceid' });
+    while ( my $adj = $adjustments->next ){
```

Comment:
Choose carefully when picking between:

`while my $thing ( $things->next ) {}` 

and

```
my @results = $things->all;
for my $thing ( @results ) {

}
```

Depending on how many results you are expecting these can have fairly different performance profiles.